hack/demo-env/setup-vcluster-env.sh

```diff
@@ -20,29 +30,176 @@ namespace() {
 	IFS=":" s=($1); echo ${s[1]}
 }
 
-trap cleanup EXIT ERR
+trap cleanup EXIT
+trap on_error ERR
+
+# check_for_openshift looks for cluster OpenShift API Resources, and if found, sets OPENSHIFT=true
+check_for_openshift() {
+	set +e
+
+	OPENSHIFT_RESOURCES=`kubectl api-resources | grep "openshift.io" | wc -l`
+
+	if [[ "$OPENSHIFT_RESOURCES" != "0" ]]; then
+		OPENSHIFT=true
+	fi
+
+	set -e
```

Comment:
@jannfis AFAICT `grep -q` as written will doing the opposite: it will set `OPENSHIFT=true` in the non-openshift cases. 

I tried using `grep -v`, and/or looking at whether any other tweaks could be used, but my bash-fu and google-fu did not reveal any obvious solutions. 

If you have a suggestion here, let me know.

Reply:
`-q` only suppresses the output of matched lines, but doesn't change exit codes:

```shell
$ echo "foo" | grep foo
foo
$ echo $?
0
$ echo "foo" | grep bar
$ echo $?
1
$ echo "foo" | grep -q foo
$ echo $?
0
$ echo "foo" | grep -q bar
$ echo $?
1
```

Reply:
@jannfis Correct, the issue is that the if statement requires an error code of 1 to execute what is inside the if block, and there is no obvious way (to my bash scripting ability) to produce an error code of 1 from `grep` in the openshift=true case.

Reply:
Found a better of doing it based on your original suggestion.

Reply:
In a shell `if` statement, an exit code of 0 is true and anything else is false :)